src/app/core/services/profile/profile.service.ts

```diff
@@ -26,30 +30,55 @@ export class ProfileService {
     };
     try {
       let data: any = await this.httpService.post(config);
+      let userDetails = await this.localStorage.getLocalData(localKeys.USER_DETAILS);
+      userDetails.user= data.user;
+      await this.localStorage.setLocalData(localKeys.USER_DETAILS, userDetails);
       this.loaderService.stopLoader();
-      this.router.navigate([CommonRoutes.TABS+'/'+CommonRoutes.PROFILE], { queryParams: formData });
+      this.router.navigate([CommonRoutes.TABS + '/' + CommonRoutes.PROFILE], { queryParams: formData });
       this.toast.showToast(data.message, "success");
     }
     catch (error) {
       this.loaderService.stopLoader();
     }
   }
-  async profileDetails() {
-    await this.loaderService.startLoader();
+  async getProfileDetailsAPI() {
     const config = {
       url: urlConstants.API_URLS.PROFILE_DETAILS,
       payload: {}
     };
     try {
       let data: any = await this.httpService.get(config);
-      this.loaderService.stopLoader();
       return data;
     }
     catch (error) {
-      this.loaderService.stopLoader();
     }
   }
-  async generateOtp(formData){
+
+  async profileDetails(): Promise<any> {
+    await this.loaderService.startLoader();
+    return new Promise((resolve) => {
+      try {
+      this.localStorage.getLocalData(localKeys.USER_DETAILS)
+        .then(async (data) => {
+          if (data?.user?.about) {
```

Comment:
do we need t0 check till about or data?.user is fine?

Reply:
yes. because in user, there will be email for all users. so data.user will be always true
and about will be there only if user has updated the data.